ledfx/tools/ts_generator.py

```
@@ -0,0 +1,841 @@
+    else:
+        fallback_effect_config_name = "BaseEffectConfig"
+        output_ts_string += f"// Fallback Base Effect Config\nexport interface {fallback_effect_config_name} {{ type?: EffectType; [key: string]: any; }}\n\n"
+        specific_effect_config_union_name = fallback_effect_config_name
+        _LOGGER.warning(
+            "No specific effect config interfaces for discriminated union."
+        )
+
+    # --- 4. Collect ALL Effect Properties for Universal Interface ---
+    all_effect_properties = {}
+    _LOGGER.info("Collecting properties for universal interface...")
+    for effect_type_str, effect_class in effect_registry.items():
```

Ruff (0.8.2)

Loop control variable `effect_type_str` not used within loop body

(B007)